internal/exec/terraform.go

```diff
@@ -379,8 +384,14 @@ func ExecuteTerraform(info schema.ConfigAndStacksInfo) error {
 	}
 
 	// Run `terraform workspace` before executing other terraform commands
-	// only if the `TF_WORKSPACE` environment variable is not set by the caller
+	// only if workspaces are enabled and the `TF_WORKSPACE` environment variable is not set by the caller
 	if info.SubCommand != "init" && !(info.SubCommand == "workspace" && info.SubCommand2 != "") {
+		// Check if workspaces are enabled
+		if !isWorkspacesEnabled(&atmosConfig, &info) {
+			// Skip workspace commands and continue
+			goto executeCommand
```

Comment:
I still have it ingrained in me that goto's are bad practice, and we should refactor so this is not necessary.